libs/components/packages/src/schematics/ng-generate/remove-compat-stylesheets/remove-compat-stylesheets.schematic.ts

```diff
@@ -0,0 +1,49 @@
+import { Rule } from '@angular-devkit/schematics';
+
+import { updateWorkspace } from '../../utility/workspace';
+
+import { Schema } from './schema';
+
+const COMPAT_STYLESHEET_REGEX = /skyux(\d+)-compat\.css$/;
+
+export default function removeCompatStylesheets(
+  options: Partial<Schema>,
+): Rule {
+  return (tree) => {
+    const belowVersion = options.belowVersion ?? Number.MAX_SAFE_INTEGER;
+    return updateWorkspace((workspace) => {
+      const compatStylesheetPaths: string[] = [];
+
+      // Remove stylesheets from the workspace config.
+      for (const project of workspace.projects.values()) {
+        for (const targetName of ['build', 'test']) {
+          const stylesheets = project.targets.get(targetName)?.options?.[
+            'styles'
+          ] as string[] | undefined;
+
+          if (stylesheets) {
+            for (let i = stylesheets.length - 1; i >= 0; i--) {
+              const stylesheet = stylesheets[i];
+
+              const check = stylesheet.match(COMPAT_STYLESHEET_REGEX);
+              if (check && parseInt(check[1]) <= belowVersion) {
```

Comment:
The logic here is "less than or equal to", but the option is `belowVersion`. Should we change the logic to `<` or the name of the option to match the logic?

How likely do you think our consumers will want to delete stylesheets less than a specific version? The feature we're going for seems to be, "delete all _outdated_ compat stylesheets", but it feels like an overcomplication to give them the ability to select the version that they consider "outdated". I still think it would be a better API to have an option that communicates "keep the current version's stylesheet" instead of asking them to provide a "less than" version. The schematic would handle the logic of which version is considered "current", and the pipeline (or consumer) executing the command would simply pass the same arguments every time. Otherwise, we'll have to include "current version" logic in the pipeline when we decide to run this command.

I imagine the `skyuxconfig.json` option would be something akin to:
```
{
  "updateSettings": {
    "includeCompatStylesheet": false
  }
}
```
And the pipeline would apply the configuration to the command like this:
```
ng g @skyux/packages:remove-compat-stylesheets --keep-current={config.updateSettings.includeCompatStylesheet}
```
The argument doesn't necessarily need to be named `--keep-current`; the flag just needs to represent a feature, instead of an internal working detail.

Reply:
Updated the comparator. This approach lets the same schematic work for the update migration and, if the config opts-out of compat stylesheets, we can call `ng g @skyux/packages:remove-compat-stylesheets`.